Gems/EMotionFX/Code/MysticQt/Source/DialogStack.cpp

```diff
@@ -498,7 +498,7 @@ namespace MysticQt
         }
         if (findPreviousMaximizedDialogNeeded)
         {
-            for (auto curDialog = AZStd::make_reverse_iterator(dialog) + 1; curDialog != m_dialogs.rend(); ++curDialog)
+            for (auto curDialog = AZStd::make_reverse_iterator(dialog); curDialog != m_dialogs.rend(); ++curDialog)
```

Comment:
Hm, this is a bug introduced by the MCore::Array -> AZStd::vector conversion. The old code did this:
```cpp
const uint32 dialogIndex = FindDialog(button);
// ... snip ...
for (int32 i = dialogIndex - 1; i >= 0; --i)
```

`dialogIndex` is now an iterator instead of an index:
```cpp
const auto dialog = AZStd::find_if(begin(mDialogs), end(mDialogs), [button](const Dialog& dialog)
{
    return dialog.mButton == button;
});
```

To do the reverse iteration, I used `make_reverse_iterator`, and *to skip the current element, I used `+ 1`*. The `+ 1` is incorrect, since:
> For a reverse iterator `r` constructed from an iterator `i`, the relationship `&*r == &*(i-1)` is always true

So the `- 1` offset in the original code is built in to the reverse view.